extensions/data-transfer/portability-data-transfer-apple/src/main/java/org/datatransferproject/datatransfer/apple/signals/AppleSignalInterface.java

```diff
@@ -0,0 +1,137 @@
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.SerializationUtils;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.datatransfer.apple.AppleBaseInterface;
+import org.datatransferproject.datatransfer.apple.AppleInterfaceFactory;
+import org.datatransferproject.datatransfer.apple.constants.AuditKeys;
+import org.datatransferproject.datatransfer.apple.constants.Headers;
+import org.datatransferproject.spi.transfer.provider.SignalRequest;
+import org.datatransferproject.spi.transfer.types.CopyException;
+import org.datatransferproject.spi.transfer.types.CopyExceptionWithFailureReason;
+import org.datatransferproject.spi.transfer.types.PermissionDeniedException;
+import org.datatransferproject.spi.transfer.types.UnconfirmedUserException;
+import org.datatransferproject.transfer.JobMetadata;
+import org.datatransferproject.types.transfer.auth.AppCredentials;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.transfer.retry.RetryStrategyLibrary;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An Interface to send the Transfer Signals to Apple.
+ */
+public class AppleSignalInterface implements AppleBaseInterface {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  protected String baseUrl;
+  protected AppCredentials appCredentials;
+  protected String exportingService;
+  protected Monitor monitor;
+  protected TokensAndUrlAuthData authData;
+
+  public AppleSignalInterface(
+      @NotNull final TokensAndUrlAuthData authData,
+      @NotNull final AppCredentials appCredentials,
+      @NotNull final Monitor monitor) {
+    this.authData = authData;
+    this.appCredentials = appCredentials;
+    this.monitor = monitor;
+    this.baseUrl = "https://datatransfer.apple.com/jobs/%s/status";
+  }
+
+  @Override
+  public String sendPostRequest(@NotNull String url, @NotNull final byte[] requestData)
+      throws IOException, CopyExceptionWithFailureReason {
+    final String correlationId = UUID.randomUUID().toString();
+    final UUID jobId = JobMetadata.getJobId();
+    monitor.info(
+        () -> "POST Request from Apple Signal Handler",
+        Headers.CORRELATION_ID,
+        correlationId,
+        AuditKeys.uri,
+        url,
+        AuditKeys.jobId,
+        jobId.toString());
+
+    HttpURLConnection con = null;
+    String responseString = null;
+    try {
+      URL signalUrl = new URL(url);
+
+      con = (HttpURLConnection) signalUrl.openConnection();
+      con.setDoOutput(true);
+      con.setRequestMethod("POST");
+      con.setRequestProperty(Headers.AUTHORIZATION.getValue(), authData.getAccessToken());
```

Comment:
Following on from my comment in `SignalHandler` - do you need to be using `authData` at all here? `authData` are credentials granted by the user to act on their behalf, and while I'm not aware of what your infra is doing on the other end of this call it seems more like the worker itself informing your infrastructure of a job starting/stopping, in which case `appCredentials` should be sufficient.

Reply:
think of `appCrentials` as a contract between the OAuth and the Client of OAuth (which is DTP). Not even the public APIs have access to that secret. The Signalling APIs are no different from public APIs. App credentials are not the right auth materials to authorize this call. Moreover, we also need to authenticate this user. Without that, DTP platform can update the status of a job of any user.

Reply:
> The Signalling APIs are no different from public APIs

For the reasons I mentioned in my earlier comment I don't believe that's the case, but if that's the implementation approach you're taking I think that's all the more reason to not include `authData` in the public API of `SignalHandler` - by not leaking that implementation detail in to the abstraction, you could implement it as part of your public API while another user could implement it through a telemetry API that uses another auth scheme.

Reply:
As above